resources/testharness.js

```diff
@@ -2750,17 +2751,27 @@
         var remaining = Math.ceil(timeout_full / interval);
         var test_this = this;
 
-        var wait_for_inner = test_this.step_func(() => {
-            if (cond()) {
+        function step(result) {
```

Comment:
I think it still makes sense to wrap this in `step_func`

Reply:
There are two call sites for `step`, and both are covered by `step_func()`. Are you proposing removing them and adding step_func here?

Reply:
It felt a little clearer to me to unconditionally wrap this function so it's clear it's part of a step whenever it's called, but what you have is also fine.